### PR TITLE
Move removeRemoteSource from RemoteTaskFactory to RemoteTask

### PR DESCRIPTION
**Pull request by wenleix**
Loose ends as discussed in https://github.com/prestodb/presto/pull/12488#discussion_r271532854 ; required by recoverable grouped execution (https://github.com/prestodb/presto/issues/12124)

### CODE REVIEW

File: presto-main/src/main/java/com/facebook/presto/server/remotetask/HttpRemoteTask.java [hunk 1]
```diff
@@ -103,9 +112,10 @@
 public final class HttpRemoteTask
         implements RemoteTask
 {
-    private static final Logger log = Logger.get(HttpRemoteTask.class);
+    private static final Logger LOG = Logger.get(HttpRemoteTask.class);
```
Comment (arhimondr) [inline presto-main/src/main/java/com/facebook/presto/server/remotetask/HttpRemoteTask.java]: There are ~160 occurrences of `Logger log` vs ~20 occurrences of `Logger LOG`. I would stick with `log`, as upper case is rather used for constants. `Logger` object is not a data object, so it is hard to call it a constant.

File: presto-main/src/main/java/com/facebook/presto/server/remotetask/HttpRemoteTask.java [hunk 2]
```diff
@@ -382,6 +396,76 @@ public synchronized void setOutputBuffers(OutputBuffers newOutputBuffers)
...
+        SettableFuture<?> future = SettableFuture.create();
+        removeRemoteSourceHelper(errorTracker, request, future);
+        return future;
+    }
+
+    private void removeRemoteSourceHelper(RequestErrorTracker errorTracker, Request request, SettableFuture<?> future)
+    {
+        errorTracker.startRequest();
+
+        FutureCallback<StatusResponseHandler.StatusResponse> callback = new FutureCallback<StatusResponseHandler.StatusResponse>() {
```
Comment (arhimondr) [inline presto-main/src/main/java/com/facebook/presto/server/remotetask/HttpRemoteTask.java]: Static import `StatusResponse`

File: presto-main/src/main/java/com/facebook/presto/server/remotetask/HttpRemoteTask.java [hunk 3]
```diff
@@ -382,6 +396,76 @@ public synchronized void setOutputBuffers(OutputBuffers newOutputBuffers)
...
+    {
+        URI remoteSourceUri = uriBuilderFrom(taskLocation)
+                .appendPath("remote-source")
+                .appendPath(remoteSourceTaskId.toString())
+                .build();
+
+        Request request = prepareDelete()
+                .setUri(remoteSourceUri)
+                .build();
+        RequestErrorTracker errorTracker = new RequestErrorTracker(
```
Comment (shixuan-fan) [inline presto-main/src/main/java/com/facebook/presto/server/remotetask/HttpRemoteTask.java]: Do we need to create tracker here or maybe we should create this tracker in `removeRemoteSourceHelper` given the only usage is to pass this variable inside?

File: presto-main/src/main/java/com/facebook/presto/server/remotetask/HttpRemoteTask.java [hunk 4]
```diff
@@ -382,6 +396,76 @@ public synchronized void setOutputBuffers(OutputBuffers newOutputBuffers)
...
+                maxErrorDuration,
+                errorScheduledExecutor,
+                "Remove exchange remote source");
+
+        SettableFuture<?> future = SettableFuture.create();
+        removeRemoteSourceHelper(errorTracker, request, future);
+        return future;
+    }
+
+    private void removeRemoteSourceHelper(RequestErrorTracker errorTracker, Request request, SettableFuture<?> future)
```
Comment (shixuan-fan) [inline presto-main/src/main/java/com/facebook/presto/server/remotetask/HttpRemoteTask.java]: This method name sounds more like it removes something called remoteSourceHelper. How about `handleRemoteSourceRemoval`?

File: presto-main/src/main/java/com/facebook/presto/server/remotetask/HttpRemoteTask.java (hunk 1 above)
Comment (wenleix) [reply presto-main/src/main/java/com/facebook/presto/server/remotetask/HttpRemoteTask.java]: cc @nezihyigitbasi

File: presto-main/src/main/java/com/facebook/presto/server/remotetask/HttpRemoteTask.java (hunk 4 above)
Comment (wenleix) [reply presto-main/src/main/java/com/facebook/presto/server/remotetask/HttpRemoteTask.java]: or `doRemoveRemoteSource`  ? ;)

File: presto-main/src/main/java/com/facebook/presto/server/remotetask/HttpRemoteTask.java (hunk 3 above)
Comment (wenleix) [reply presto-main/src/main/java/com/facebook/presto/server/remotetask/HttpRemoteTask.java]: @shixuan-fan : Remember `removeRemoteSourceHelper` recursively calls itself, so we don't want to create `RequestErrorTracker` inside it :)

Comment (shixuan-fan) [reply presto-main/src/main/java/com/facebook/presto/server/remotetask/HttpRemoteTask.java]: Cool. Didn't see that.

**Pull request merged**